### ISSUE TITLE
plugin/kubernetes: deprecate resyncperiod

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#2911]
from 1.5.0 changelog:
~~~
The resyncperiod option in kubernetes is deprecated and will be made a no-op in 1.6.0 and removed in 1.7.0.
~~~
Next release is likely 1.6.0 so we can start doing this.

**Comment by sharmaansh21** [comment]
Get rid of `resyncperiod` from everywhere or they have the equivalent option?

**Comment by chrisohaver** [comment]
The aim is to remove the option from the _kubernetes_ plugin with no equivalent replacement.
The option is already deprecated (as of 1.5.0).
In 1.6.0, it will be ignored if used in the Corefile.
In a following release it will cause an error if used in the Corefile.

IIUC, [this discussion](https://groups.google.com/forum/#!topic/kubernetes-sig-api-machinery/PbSCXdLDno0) reveals that `resyncperiod` is not useful in the context of how the _kubernetes_ plugin uses the API (a read-only watcher).  Furthermore, the function of the option is not what one might expect in this context, i.e. resync != relist.

**Comment by sharmaansh21** [comment]
I can work on the deprecation part on the coredns side :bowtie:

**Comment by sharmaansh21** [comment]
If I am going in the right direction, we need to wait for https://github.com/kubernetes/client-go/blob/master/tools/cache/controller.go#L52 to be removed first?
Used here https://github.com/coredns/coredns/blob/master/plugin/kubernetes/object/informer.go#L20

**Comment by chrisohaver** [comment]
@sharmaanshul21,  Just pass `0`.  The option isn't going to be removed from the kubernetes API client. IIRC, `0` disables the resync.

```
FullResyncPeriod: 0,
```

**Comment by miekg** [comment]
this has been merged and will go out in 1.6.0.
